### DISCUSSION TITLE
Hot Chocolate: Autogenerated type for IDictionary becomes List

### DISCUSSION THREAD

**Discussion by nurbolnygmanov** [ChilliCream/hotchocolate#7268] [Q&A] [answered]
TL.DR.
Is it ok if IDictionary type gets converted into List of key value pairs during type generation?
Is it expected behavior for hotchocolate to generate Lists in case of Dictionary usage?

I have class with dictionary and query as below
```
public class Product
{
    public string Name { get; set; }
    public Dictionary<string, string> Attributes { get; set; }
}

```
```

public class Query
{
    public Product GetProduct() =>
        new Product
        {
            Name = "Chocolate Bar",
            Attributes = new Dictionary<string, string>
            {
                { "Flavor", "Dark" },
                { "Weight", "100g" }
            }
        };
}
```

is it ok that above setup generates below type, and what is the reason behind it?
```

type Product {
  name: String
  attributes: [Attribute] //⚠️attribute is List here
}

type Attribute {
  key: String
  value: String
}

type Query {
  product: Product
}
```

and retuned json is 
```
{
  "data": {
    "product": {
      "name": "Chocolate Bar",
      "attributes": [
        { "key": "Flavor", "value": "Dark" },
        { "key": "Weight", "value": "100g" }
      ]
    }
  }
}
```

**Comment by eduardocp** [comment, marked as answer]
Yes. This is the expected behavior. Dictionaries uses `KeyValuePair<TKey, TValue>` and this is the structure that is converted. I think you can customize this by creating [Custom Converters](https://chillicream.com/docs/hotchocolate/v13/defining-a-schema/scalars/#custom-converters)
